### ISSUE TITLE
[Bug]: split-by-size not respecting maxSize

### ISSUE THREAD

**Issue by CristianPi** [web-infra-dev/rspack#7603]
### Version

```sh
System:
    OS: Linux 6.10 Fedora Linux 40 (Workstation Edition)
    CPU: (4) x64 Intel(R) Core(TM) i5-7600K CPU @ 3.80GHz
    Memory: 7.80 GB / 19.42 GB
    Container: Yes
    Shell: 5.2.26 - /bin/bash
  Browsers:
    Chrome: 127.0.6533.119
  npmPackages:
    @rsbuild/core: 1.0.1-beta.14 => 1.0.1-beta.14 
    @rsbuild/plugin-react: 1.0.1-beta.14 => 1.0.1-beta.14 
    @rsbuild/plugin-sass: 1.0.1-beta.14 => 1.0.1-beta.14 
    @rsbuild/plugin-svgr: 1.0.1-beta.14 => 1.0.1-beta.14
```

### Details

```js

    performance: {
        chunkSplit: {
            strategy: 'split-by-size',
            // MAX 6 MB for lambda
            minSize: 1000 * 1000,
            maxSize: (1000 * 1000 * 6) - 1,
        },
    },

```

![Screenshot_20240816_100303](https://github.com/user-attachments/assets/0b0b0460-5f0e-4ccd-a1cc-a73df794d24f)

~8MB of size, in the code a .json of around 4MB is imported, i expected that size to be imported separately.

If i remove the .json the maxSize is respected

![Screenshot_20240816_100828](https://github.com/user-attachments/assets/831079b4-5658-490f-a451-f9dbe464d7ff)

### Reproduce link

https://github.com/CristianPi/rs-build-bugs

### Reproduce Steps

rsbuild build

check files sizes to be less than maxSize

**Comment by CPunisher** [comment]
I think this is expected according to https://webpack.js.org/plugins/split-chunks-plugin/#splitchunksmaxsize. I get the same results with webpack.
> maxSize is only a hint and could be violated when modules are bigger than maxSize or splitting would violate minSize.

So it could be better to split chunks manually by `import()` or cache group.

**Comment by JSerFeng** [comment]
Fixed in #8442
